improver_tests/uv_index/test_uv_index.py

```diff
@@ -70,46 +61,71 @@ def test_basic(self):
         default scaling factor. Make sure the output is a cube
         with the expected data."""
         scale_factor = 1.0
-        expected = np.array([[0.3, 0.3, 0.3], [0.3, 0.3, 0.3]], dtype=np.float32)
-        result = calculate_uv_index(self.cube_uv_up, self.cube_uv_down, scale_factor)
+        expected = self.cube_uv_down.data.copy()
+        result = calculate_uv_index(self.cube_uv_down, scale_factor)
         self.assertArrayEqual(result.data, expected)
 
     def test_scale_factor(self):
         """ Test the uv calculation works when changing the scale factor. Make
         sure the output is a cube with the expected data."""
-        expected = np.array([[3.0, 3.0, 3.0], [3.0, 3.0, 3.0]], dtype=np.float32)
-        result = calculate_uv_index(self.cube_uv_up, self.cube_uv_down, scale_factor=10)
+        expected = np.ones_like(self.cube_uv_down.data, dtype=np.float32)
+        result = calculate_uv_index(self.cube_uv_down, scale_factor=10)
         self.assertArrayEqual(result.data, expected)
 
     def test_metadata(self):
         """ Tests that the uv index output has the correct metadata (no units,
         and name = ultraviolet index)."""
-        result = calculate_uv_index(self.cube_uv_up, self.cube_uv_down)
+        result = calculate_uv_index(self.cube_uv_down)
         self.assertEqual(str(result.standard_name), "ultraviolet_index")
         self.assertIsNone(result.var_name)
         self.assertIsNone(result.long_name)
         self.assertEqual((result.units), Unit("1"))
 
-    def test_diff_units(self):
-        """Tests that a ValueError is raised if the input uv files have
-        different units. """
-        msg = "The input uv files do not have the same units."
-        with self.assertRaisesRegex(ValueError, msg):
-            calculate_uv_index(self.cube_uv_up, self.cube_diff_units, scale_factor=1.0)
-
     def test_badname_down(self):
         """Tests that a ValueError is raised if the input uv down
         file has the wrong name. """
         msg = "The radiation flux in UV downward"
```

Comment:
This test actually needs a slightly longer expected, as there are two exceptions that both raise an error starting with "The radiation flux in UV downward", meaning this test cannot currently differentiate between them and check the correct exception has been triggered.